homeassistant/components/thread/config_flow.py

```diff
@@ -12,8 +13,18 @@ class ThreadConfigFlow(ConfigFlow, domain=DOMAIN):
 
     VERSION = 1
 
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Set up because the user has border routers."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
+        return self.async_create_entry(title="Thread", data={})
+
     async def async_step_import(
         self, import_data: dict[str, str] | None = None
     ) -> FlowResult:
         """Set up by import from async_setup."""
+        if self._async_current_entries():
+            return self.async_abort(reason="already_configured")
```

Comment:
We don't need this, we check it in `async_setup` before starting the import flow.
Maybe add a comment, it's not the first time it's added.